evalml/data_checks/target_leakage_data_check.py

```diff
@@ -1,69 +1,72 @@
 """Data check that checks if any of the features are highly correlated with the target by using mutual information or Pearson correlation."""
+from woodwork.config import CONFIG_DEFAULTS
+
 from evalml.data_checks import (
     DataCheck,
     DataCheckActionCode,
     DataCheckActionOption,
     DataCheckMessageCode,
     DataCheckWarning,
 )
-from evalml.utils.woodwork_utils import infer_feature_types, numeric_and_boolean_ww
+from evalml.utils.woodwork_utils import infer_feature_types
 
 
 class TargetLeakageDataCheck(DataCheck):
-    """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation.
+    """Check if any of the features are highly correlated with the target by using mutual information, Pearson correlation, and other correlation metrics.
 
-    If `method='mutual'`, this data check uses mutual information and supports all target and feature types.
-    Otherwise, if `method='pearson'`, it uses Pearson correlation and only supports binary with numeric and boolean dtypes.
-    Pearson correlation returns a value in [-1, 1], while mutual information returns a value in [0, 1].
+    If `method='mutual_info'`, this data check uses mutual information and supports all target and feature types.
+    Other correlation metrics only support binary with numeric and boolean dtypes, and return a value in [-1, 1], while mutual information returns a value in [0, 1].
+    Correlation metrics available can be found in Woodwork's `dependence_dict method <https://woodwork.alteryx.com/en/stable/generated/woodwork.table_accessor.WoodworkTableAccessor.dependence_dict.html#woodwork.table_accessor.WoodworkTableAccessor.dependence_dict>`_.
 
     Args:
         pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
-        method (string): The method to determine correlation. Use 'mutual' for mutual information, otherwise 'pearson' for Pearson correlation. Defaults to 'mutual'.
+        method (string): The method to determine correlation. Use 'max' for the maximum correlation, or for specific correlation metrics, use their name (ie 'mutual_info' for mutual information, 'pearson' for Pearson correlation, etc).
+            Excludes 'all'. Defaults to 'mutual_info'.
     """
 
-    def __init__(self, pct_corr_threshold=0.95, method="mutual"):
+    def __init__(self, pct_corr_threshold=0.95, method="mutual_info"):
         if pct_corr_threshold < 0 or pct_corr_threshold > 1:
             raise ValueError(
                 "pct_corr_threshold must be a float between 0 and 1, inclusive.",
             )
-        if method not in ["mutual", "pearson"]:
-            raise ValueError(f"Method '{method}' not in ['mutual', 'pearson']")
+        methods = CONFIG_DEFAULTS["correlation_metrics"]
+        if method not in methods:
+            raise ValueError(f"Method '{method}' not in {methods}")
```

Comment:
can we add: `raise ValueError(f"Method '{method}' not in available correlation methods. Available methods include {methods}")`